runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -135,6 +158,112 @@ private void sendNotification(
 		}
 	}
 
+	private void sendCustomNotification(
+		final String title,
+		final String message,
+		final TrayIcon.MessageType type,
+		final String subtitle)
+	{
+		final int width = 360;
+		final int height = 100;
+		final int padding = 50;
+
+		JFrame parentWindow = clientUI.get().getFrame();
+		Rectangle display = parentWindow.getGraphicsConfiguration().getBounds();
+
+		JFrame window = new JFrame();
+		window.setUndecorated(true);
+		window.setIconImage(ClientUI.ICON);
+		window.setFocusableWindowState(false);
+		window.setBounds(
+			display.x + display.width - width - padding,
+			display.y + display.height - height - padding,
+			width,
+			height
+		);
+		((JPanel) window.getContentPane()).putClientProperty(SubstanceSynapse.COLORIZATION_FACTOR, 1.0);
+
+		window.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent e)
+			{
+				window.setVisible(false);
+				window.dispose();
+			}
+		});
+		new Timer(5000, e ->
+		{
+			window.setVisible(false);
+			window.dispose();
+		}).start();
+
+		float[] hsbs = new float[3];
+		Color stock = window.getContentPane().getBackground();
+		Color.RGBtoHSB(stock.getRed(), stock.getGreen(), stock.getBlue(), hsbs);
+		final long start = System.currentTimeMillis();
+		Timer anim = new Timer(20, e ->
```

Comment:
I thought creating threads like this (timer is creating thread i guess) is generally disliked

Reply:
There is no other way that I know of to do this.

Reply:
Would be nice to extend current scheduler implementation to work outside of plugins probably, I needed that too before

Reply:
Even if we did that that wouldn't help because I need to cancel

Reply:
Its very unnoticeable if I don't animate it

Reply:
Also I don't think this creates a thread, because it runs on the EDT